src/ReTestItems.jl

```diff
@@ -245,6 +255,10 @@ will be run.
   Defaults to the value passed to the `failfast` keyword.
   If a `@testitem` sets its own `failfast` keyword, then that takes precedence.
   Note that the `testitem_failfast` keyword only takes effect in Julia v1.9+ and is ignored in earlier Julia versions.
+- `failures_first::Bool=false`: if `true`, first runs test items that failed the last time
+  they ran, followed by new test items, followed by test items that passed the
+  last time they ran. This is currently only supported when `nworkers=0`.
+  Can also be set using the `RETESTITEMS_FAILURES_FIRST` environment variable.
```

Comment:
Why isn't this supported with multiple workers? It seems to me the coordinator controls the order of the queue and knows the result of the test item.

Reply:
that's a very different implementation, because it flips where state is stored. right now each testitem knows what number it is and if it's available (or already taken), and we simply ask for the next available one, and there's nowhere that know the state of each worker. whereas in this proposal there would be a central manager. So that's a big change, and i don't think this feature requires it.